pyatlan/model/fluent_search.py

```diff
@@ -0,0 +1,391 @@
+        return Referenceable.SUPER_TYPE_NAMES.eq(one_of.__name__)
+
+    @staticmethod
+    def tagged(
+        with_one_of: Optional[list[str]] = None, directly: bool = False
+    ) -> Query:
+        """
+        Returns a query that will only match assets that have at least one of the Atlan tags
+        provided. This will match irrespective of the Atlan tag being directly applied to the
+        asset, or if it was propagated to the asset.
+
+        :param with_one_of: human-readable names of the Atlan tags
+        :param directly: when True, the asset must have at least one Atlan tag directly assigned, otherwise
+                         even propagated tags will suffice
+        :returns: a query that will only match assets that have at least one of the Atlan tags provided
+        """
+        from pyatlan.cache.atlan_tag_cache import AtlanTagCache
+
+        values: list[str] = []
+        if with_one_of:
+            for name in with_one_of:
+                if tag_id := AtlanTagCache.get_id_for_name(name):
+                    values.append(tag_id)
+        if directly:
+            if values:
+                return FluentSearch(
+                    wheres=[Referenceable.ATLAN_TAGS.within(values)]
+                ).to_query()
+            return FluentSearch(wheres=[Referenceable.ATLAN_TAGS.exists()]).to_query()
+        if values:
+            return FluentSearch(
+                where_somes=[
+                    Referenceable.ATLAN_TAGS.within(values),
+                    Referenceable.PROPAGATED_ATLAN_TAGS.within(values),
+                ],
+                _min_somes=1,
+            ).to_query()
+        return FluentSearch(
+            where_somes=[
+                Referenceable.ATLAN_TAGS.exists(),
+                Referenceable.PROPAGATED_ATLAN_TAGS.exists(),
+            ],
+            _min_somes=1,
+        ).to_query()
+
+    @staticmethod
+    def assigned_term(qualified_names: Optional[list[str]] = None) -> Query:
+        """
+        Returns a query that will only match assets that have at least one term assigned.
+        (If a list of qualified_names is specified, the assets that match must have at least
+        one term assigned from within the list of qualified_names.)
+
+        :param qualified_names: the qualified_names of the terms
+        :returns: a query that will only match assets that have at least one term assigned
+        """
+        if qualified_names:
+            return Referenceable.ASSIGNED_TERMS.within(qualified_names)
+        return Referenceable.ASSIGNED_TERMS.exists()
+
+    def __init__(
+        self,
+        wheres: Optional[list[Query]] = None,
+        where_nots: Optional[list[Query]] = None,
+        where_somes: Optional[list[Query]] = None,
+        _min_somes: int = 1,
+    ):
+        self.wheres = wheres
+        self.where_nots = where_nots
+        self.where_somes = where_somes
+        self._min_somes = _min_somes
+
+    def _clone(self) -> "CompoundQuery":
```

Comment:
The convention in python is to override the __copy__ and __deepcopy__ methods. See [copy- Shallow and deep copy operations](https://docs.python.org/3/library/copy.html) and [Shallow and Deep Copy in Python and How to Use __copy__](https://thepythoncodingbook.com/2022/08/10/shallow-and-deep-copy-in-python/) Maybe just add a TODO comment to refactor in the future.